### ISSUE TITLE
The extension 'amp-wistia-player' was found on this page, but is unused. Please remove this extension.

### ISSUE THREAD

**Issue by vagu71** [ahmedkaludi/accelerated-mobile-pages#3587]
Even when there is no wistia content in the amp builder for a given page, this error shows on when viewing the amp version of the page: 

"The extension 'amp-wistia-player' was found on this page, but is unused. Please remove this extension."

FYI, the non amp version has got WISTIA videos. 

The code for wistia embed is in the file \plugins\accelerated-mobile-pages\includes\vendor\amp\includes\embeds\class-amp-wistia-embed.php.

**Comment by vagu71** [comment]
I changed the render function to the following code: 

`public function render( $args ) {
		$args = wp_parse_args( $args, array(
			'video_id' => false,
		) );

		if ( empty( $args['video_id'] ) ) {
			return AMP_HTML_Utils::build_tag( 'a', array( 'href' => esc_url( $args['url'] ), 'class' => 'amp-wp-embed-fallback' ), esc_html( $args['url'] ) );
		}

		$this->did_convert_elements = true;
		$attrs = array(
				'data-media-hashed-id' => $args['video_id'],
				'width' => $this->args['width'],
				'height' => $this->args['height'],
				'layout' => 'responsive',
			);
		$attrs = ampforwp_amp_consent_check($attrs);
		$this->did_convert_elements = false;
		return AMP_HTML_Utils::build_tag(
			'amp-wistia-player',$attrs);
	}`

Now the wistia js does not render on the frontend, the wistia iframe video plays good. And the error is also removed. Thoughts?

**Comment by ahmedkaludi** [comment]
in which scenario does that amp-wistia-player renders?

**Comment by vagu71** [comment]
Here's what's happening: 

1. The page non amp content is being built out using wpbakery visual composer. In that, the wistia videos are embedded using the following code (the following code is added in a vc text block): 
`<script src="https://fast.wistia.com/embed/medias/qb6cewfcyn.jsonp" async></script><script src="https://fast.wistia.com/assets/external/E-v1.js" async></script> 
<div class="wistia_responsive_padding" style="padding: 56.25% 0 0 0; position: relative;"> 
<div class="wistia_responsive_wrapper" style="height: 100%; left: 0; position: absolute; top: 0; width: 100%;"><span class="wistia_embed wistia_async_qb6cewfcyn popover=true popoverAnimateThumbnail=true videoFoam=true" style="display: inline-block; height: 100%; position: relative; width: 100%;"> </span></div> 
</div>`

2. Then in the amp builder section, the video is added using the video component of the amp theme framework. And then in the type dropdown, the value Embed Type is chosen and then in the Embed video iframe, the following code is added: 
`<iframe src="https://fast.wistia.net/embed/iframe/7pqit19iol?videoFoam=true" title="Why replacing a non-working part may NOT fix the problem" allowtransparency="true" frameborder="0" scrolling="no" class="wistia_embed" name="wistia_embed" allowfullscreen mozallowfullscreen webkitallowfullscreen oallowfullscreen msallowfullscreen width="720" height="305" align="center"></iframe>`

On the frontend, this error comes: 
 The extension 'amp-wistia-player' was found on this page, but is unused. Please remove this extension.

Solution: In the file \plugins\accelerated-mobile-pages\includes\vendor\amp\includes\embeds\class-amp-wistia-embed.php, we changed the function render to: 

`public function render( $args ) {
$args = wp_parse_args( $args, array(
'video_id' => false,
) );
if ( empty( $args['video_id'] ) ) {
return AMP_HTML_Utils::build_tag( 'a', array( 'href' => esc_url( $args['url'] ), 'class' => 'amp-wp-embed-fallback' ), esc_html( $args['url'] ) );
}

$this->did_convert_elements = true;
$attrs = array(
	'data-media-hashed-id' => $args['video_id'],
	'width' => $this->args['width'],
	'height' => $this->args['height'],
	'layout' => 'responsive',
);
$attrs = ampforwp_amp_consent_check($attrs);
$this->did_convert_elements = false;
return AMP_HTML_Utils::build_tag(
'amp-wistia-player',$attrs);
}`

Basically, '$this->did_convert_elements = false;' was added on line number 86.

This solved the issue. The videos load good, and also the error is now gone.

**Comment by Zabi09** [comment]
Pushed in 0.9.98.8
